### PR TITLE
[ROCKETMQ-273] return an expression when a function has no write operations.

### PR DESCRIPTION
**Pull request by kevin-better**
what about return an expression when a function has no write operations?

### CODE REVIEW

**Comment by lindzh** [comment]
Thanks for contribution.
Please add a jira issue and modify the pr title to make it complete

**Comment by kevin-better** [comment]
@lindzh  ok

**Comment by kevin-better** [comment]
@lindzh this is my first time to create an issue on jira,what will I do on next ?

**Comment by vsair** [comment]
what is the purpose of this issue?  it confused me.

**Comment by kevin-better** [comment]
@vsair change style of this code .
eg: if(true) return true; change to return true;

File: store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java [hunk 1]
```diff
@@ -1094,34 +1090,15 @@ private boolean checkInDiskByCommitOffset(long offsetPy, long maxOffsetPy) {
...
-                return true;
-            }
-
-            if (messageTotal > this.messageStoreConfig.getMaxTransferCountOnMessageInMemory() - 1) {
-                return true;
-            }
-        }
-
-        return false;
+        return    (0 != bufferTotal && 0 != messageTotal)
```
Comment (lizhanhui) [inline store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java]: This one looks too complex to understand...I suggest to keep the original look-and-feel.

Comment (kevin-better) [inline store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java]: ok , only alter function isOSPageCacheBusy

**Comment by kevin-better** [comment]
Travis CI build log ,is there something wrong ? : 

[https://travis-ci.org/apache/incubator-rocketmq/jobs/267873150](https://travis-ci.org/apache/incubator-rocketmq/jobs/267873150)

$ mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V
Error: JAVA_HOME is not defined correctly.
  We cannot execute /usr/lib/jvm/java-7-oracle/bin/java
The command "eval mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V " failed. Retrying, 2 of 3.
Error: JAVA_HOME is not defined correctly.
  We cannot execute /usr/lib/jvm/java-7-oracle/bin/java
The command "eval mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V " failed. Retrying, 3 of 3.
Error: JAVA_HOME is not defined correctly.
  We cannot execute /usr/lib/jvm/java-7-oracle/bin/java
The command "eval mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V " failed 3 times.
The command "mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V" failed and exited with 1 during .
Your build has been stopped.

**Comment by dongeforever** [comment]
@kevin-better This PR has been merged. You may close it now.

**Comment by kevin-better** [comment]
@dongeforever  ok

File: store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java [hunk 2]
```diff
@@ -1094,34 +1090,15 @@ private boolean checkInDiskByCommitOffset(long offsetPy, long maxOffsetPy) {
...
-            }
-
-            if (messageTotal > this.messageStoreConfig.getMaxTransferCountOnMessageInMemory() - 1) {
-                return true;
-            }
-        }
-
-        return false;
+        return   (0 != bufferTotal && 0 != messageTotal)
+                &&(
```
Comment (vongosling) [inline store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java]: Human readable problem

Comment (kevin-better) [reply store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java]: It has been restored

**Pull request closed without merging**